src/nvim/api/vim.c

```diff
@@ -778,6 +782,42 @@ void nvim_set_option_value(String name, Object value, Dict(option) *opts, Error
     return;
   }
 
+  int opt_type = SREQ_GLOBAL;
+  void *to = NULL;
+
+  if (opts->win.type == kObjectTypeInteger) {
+    scope = OPT_LOCAL;
+    opt_type = SREQ_WIN;
+    to = find_window_by_handle((int)opts->win.data.integer, err);
+  } else if (HAS_KEY(opts->win)) {
+    api_set_error(err, kErrorTypeValidation, "invalid value for key: win");
+    return;
+  }
+
+  if (opts->buf.type == kObjectTypeInteger) {
+    scope = OPT_LOCAL;
+    opt_type = SREQ_BUF;
+    to = find_buffer_by_handle((int)opts->buf.data.integer, err);
+  } else if (HAS_KEY(opts->buf)) {
+    api_set_error(err, kErrorTypeValidation, "invalid value for key: buf");
+    return;
+  }
+
+  if (HAS_KEY(opts->scope) && HAS_KEY(opts->win)) {
```

Comment:
This should be allowed.

Reply:
Given how it's implemented, scope will just be ignored.

Reply:
which means it should be fixed. all of `setglobal`, `set` and `setlocal` are meaningful for a specific window.

Reply:
My understanding of all this is still incomplete, so I'll need a little extra help if that's ok? Is my comment on the other PR at least correct?